### ISSUE TITLE
Passing a date as a `MomentInputObject` to moment is off by one. Months start at 0?

### ISSUE THREAD

**Issue by SeedyROM** [moment/moment#6009]
**Describe the bug**
When you pass an object like `{ day: 5, months: 4, year: 2022 }` you get back a May 5th 2022. Not April... why?

**To Reproduce**
Here.
https://codesandbox.io/s/vigorous-snyder-d8gxym?file=/src/index.js

**Expected behavior**
Months should not start at 0.

Please run the following code in your environment and include the output:
```
console.log((new Date()).toString())
console.log((new Date()).toLocaleString())
console.log((new Date()).getTimezoneOffset())
console.log(navigator.userAgent)
console.log(moment.version)
```

```
Sun May 29 2022 22:00:52 GMT-0700 (Pacific Daylight Time) 
5/29/2022, 10:00:52 PM 
420
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.54 Safari/537.36 
2.29.3 
```

**Comment by sciffany** [comment]
https://momentjs.com/docs/#/get-set/month/

**Comment by SeedyROM** [comment]
> https://momentjs.com/docs/#/get-set/month/

Yeah I get that after reading the docs, but why doesn't the day of the month follow the same pattern? Same with the year?

So if I gave the object `{ day: 0, month: 0, year: 2009 }` I'd get `2010-01-01`?

And if it's about calendar differences with years/months, why would it correctly parse for the Gregorian calendar?
